barretenberg/cpp/src/barretenberg/benchmark/mega_memory_bench/mega_memory.bench.cpp

```diff
@@ -0,0 +1,344 @@
+
+        // secp256k1 generator slices
+        plookup_read::get_lookup_accumulators(MultiTableId::SECP256K1_XLO, byte, 0, false);
+        plookup_read::get_lookup_accumulators(MultiTableId::SECP256K1_XHI, byte, 0, false);
+        plookup_read::get_lookup_accumulators(MultiTableId::SECP256K1_YLO, byte, 0, false);
+        plookup_read::get_lookup_accumulators(MultiTableId::SECP256K1_YHI, byte, 0, false);
+        plookup_read::get_lookup_accumulators(MultiTableId::SECP256K1_XYPRIME, byte, 0, false);
+        plookup_read::get_lookup_accumulators(MultiTableId::SECP256K1_XLO_ENDO, byte, 0, false);
+        plookup_read::get_lookup_accumulators(MultiTableId::SECP256K1_XHI_ENDO, byte, 0, false);
+        plookup_read::get_lookup_accumulators(MultiTableId::SECP256K1_XYPRIME_ENDO, byte, 0, false);
+
+        // blake xor
+        plookup_read::get_lookup_accumulators(MultiTableId::BLAKE_XOR, left, right, true);
+        plookup_read::get_lookup_accumulators(MultiTableId::BLAKE_XOR_ROTATE_16, left, right, true);
+        plookup_read::get_lookup_accumulators(MultiTableId::BLAKE_XOR_ROTATE_8, left, right, true);
+        plookup_read::get_lookup_accumulators(MultiTableId::BLAKE_XOR_ROTATE_7, left, right, true);
+
+        // keccak tests trigger
+        // SharedShiftedVirtualZeroesArray ... Assertion `(index >= start_ && index < end_)' failed.
+        // plookup_read::get_lookup_accumulators(MultiTableId::KECCAK_THETA_OUTPUT, left, right, true);
+        // plookup_read::get_lookup_accumulators(MultiTableId::KECCAK_CHI_OUTPUT, left, right, true);
+        // plookup_read::get_lookup_accumulators(MultiTableId::KECCAK_FORMAT_INPUT, left, right, true);
+        // plookup_read::get_lookup_accumulators(MultiTableId::KECCAK_FORMAT_OUTPUT, left, right, true);
+        // plookup_read::get_lookup_accumulators(MultiTableId::KECCAK_NORMALIZE_AND_ROTATE, left, right, true);
+    }
+}
+
+void fill_trace(State& state, TraceSettings settings)
+{
+    Builder builder;
+    builder.blocks.set_fixed_block_sizes(settings);
+
+    fill_ecc_op_block(builder);
+    fill_pub_inputs_block(builder);
+    fill_databus_blocks(builder);
+    fill_delta_range_block(builder);
+    fill_arithmetic_block(builder); // must come after fill_delta_range_block
+    fill_elliptic_block(builder);
+    fill_aux_block(builder);
+    fill_poseidon2_external_block(builder);
+    fill_poseidon2_internal_block(builder);
+    fill_lookup_block(builder);
+
+    {
+        // finalize doesn't populate public inputs block, so copy to verify that the block is being filled well.
+        // otherwise the pk construction will overflow the block
+        // alternative: add to finalize or add a flag to check whether PIs have already been populated
+        auto builder_copy = builder;
+        builder_copy.finalize_circuit(/* ensure_nonzero */ false);
+        DeciderProvingKey::Trace::populate_public_inputs_block(builder_copy);
```

Comment:
One other thing. If you just construct a DeciderProvingKey from this builder_copy, it will now automatically compute the overflow / whether or not there is overflow. The print isn't as detailed as this one because I didn't add the labels (as you now have) but it could be made to be similar to what you have here (and probably should be for the sake of dev experience)